internal/manifests/datasource/grafana.go

```diff
@@ -0,0 +1,38 @@
+package datasource
+
+import (
+	grafanav1 "github.com/grafana-operator/grafana-operator/v5/api/v1beta1"
+	configv1alpha1 "github.com/grafana/tempo-operator/apis/config/v1alpha1"
+	"github.com/grafana/tempo-operator/internal/manifests/manifestutils"
+)
+
+// Datasource creates a Datasource for Grafana Tempo.
+func Datasource(featureGates configv1alpha1.FeatureGates, namespace string) *grafanav1.GrafanaDatasource {
```

Comment:
```suggestion
func Datasource(params manifestutils.Params) *grafanav1.GrafanaDatasource {
```
I'd suggest using `manifestutils.Params` here, to have access to the Tempo spec (containg the namespace) and all feature gates like with the other manifests, e.g. https://github.com/grafana/tempo-operator/blob/bb9d581a4be99b3d551f449c68374911a412e703/internal/manifests/distributor/distributor.go#L21

Reply:
Agree, will change.